### DISCUSSION TITLE
Code for different browsers? (non-coder)

### DISCUSSION THREAD

**Discussion by sugarfreejaz** [google/model-viewer#4588] [Q&A]
Adding AR option to product pages, no scene, just View in your space option.   Very basic, no lighting, orbit etc, only specifying place on wall.

Works on Safari, Firefox, but will not work Edge at all and in chrome it shifts the contents of entire page it is on to the left , but will go to AR mode.  That shift could be my lack of coding knowledge.

Using edge it shows the file name only and does nothing.

I do not know anything about coding.    I am pasting code into html page where i think it should go, save, refresh try again, repeat.

Is there code you need to add for diff browsers so it will work?

**Comment by elalish** [comment]
I think the Edge problem is their bug: https://github.com/google/model-viewer/issues/4571

In general, no, you shouldn't need to do special stuff for different browsers - we take care of that for you, to the extent possible. But, the web is complex and something outside of MV could easily be hitting inconsistent behavior.

**Comment by sugarfreejaz** [comment]
found the edge bug post. thanks.

well I am at a loss to get the 3 lines of code to work!!  all i want is the little box icon to appear on mobile devices, and when tapped the object appears in AR.  

the icon appears, and then that's it.  link dead.

**Reply by elalish** [reply]
Is this just on Edge? It's helpful when you report to share a link we can try and specify exactly which devices & browsers it's having trouble on. Edge will be a no-go until Microsoft fixes the bug on their side, which will probably be when they update Chromium versions.

**Reply by sugarfreejaz** [reply]
i do not really care about Edge, traffic is zero from edge.    chrome, FF and safari is my goal.  below is the script.  being put into a template of a product page: www.uncarvedblock.com.au/xxlj41  (in description tab)  i want the View on Your Wall button to appear below or above picture, but cannot get it to work in any browser. 

have a dynamic version that inserts product name and URL to object files which are being served with correct MIME.   someone has tried to explain the module inserts some html, css etc and that could be the issue.   

<script type="module" src="https://ajax.googleapis.com/ajax/libs/model-viewer/3.3.0/model-viewer.min.js"></script>

<model-viewer alt="A 3D model of UB-XXL-J41" ar="" ar-placement="wall" ar-scale="fixed" camera-controls="" ios-src="https://s3.ap-southeast-2.amazonaws.com/uncarvedblock.com.au/UB-XXL-J41.usdz" src="https://s3.ap-southeast-2.amazonaws.com/uncarvedblock.com.au/UB-XXL-J41.glb" touch-action="pan-y" xr-environment="">

<button slot="ar-button" style="background-color: #f1f1f1; border-radius: 0px; border: none; position: bottom; top: 0px; right: 0px; ">View On Your Wall!</button> 

</model-viewer>

**Reply by elalish** [reply]
Oh, you want AR with no 3D? See https://modelviewer.dev/docs/faq.html#entrydocs-general-questions-no3d

You said earlier "Works on Safari, Firefox", so I'm not sure what's not working.

**Reply by sugarfreejaz** [reply]
ok got it working.    it works for me on safari, FF and chrome on iphone.  seems very hit and miss, friends telling me it won't work for them.  

tech need to catch up with your code i guess?
